safe/gui/tools/options_dialog.py

```diff
@@ -302,34 +322,57 @@ def save_state(self):
             self.leUserDirectoryPath.text())
         index = self.earthquake_function.currentIndex()
         value = self.earthquake_function.itemData(index)
-        set_setting('earthquake_function', value)
+        set_setting('earthquake_function', value, qsettings=self.settings)
+
+        currency_index = self.currency_combo_box.currentIndex()
+        currency_key = self.currency_combo_box.itemData(currency_index)
+        set_setting('currency', currency_key, qsettings=self.settings)
 
         # Save InaSAFE default values
         self.save_default_values()
 
     def accept(self):
         """Method invoked when OK button is clicked."""
-        if not self.is_good_age_ratios():
-            display_warning_message_box(
-                self,
-                tr('Wrong Sum Age Ratio'),
-                tr('You have set age ratio whose sum is not equal to 1. '
-                   'Please fix it in the <b>Global Default</b> tab before you '
-                   'can save it.'))
-            return
         self.save_state()
-        # FIXME: Option dialog should be independent from dock.
-        if self.dock:
-            self.dock.read_settings()
-        self.close()
+        super(OptionsDialog, self).accept()
+
+    # noinspection PyPep8Naming
+    @pyqtSignature('int')  # prevents actions being handled twice
+    def on_earthquake_function_currentIndexChanged(self, current_index):
+        """Auto-connect slot activated when earthquake model is changed."""
+        self.label_earthquake_model()
+        model = EARTHQUAKE_FUNCTIONS[current_index]
+        notes = ''
+        for note in model['notes']:
+            notes += note + '\n\n'
+
+        citations = ''
+        for citation in model['citations']:
+            citations += citation['text'] + '\n\n'
+
+        text = tr('Description:\n\n%s\n\n'
+            'Notes:\n\n%s\n\n'
+            'Citations:\n\n%s') % (
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->